go-controller/hybrid-overlay/cmd/hybrid-overlay-node/hybrid-overlay-node.go

```diff
@@ -103,12 +104,7 @@ func runHybridOverlay(ctx *cli.Context) error {
 	}
 
 	f.Start(stopChan)
-	go func() {
-		err := n.Run(stopChan)
-		if err != nil {
-			klog.Error(err)
-		}
-	}()
+	n.Run(stopChan, &sync.WaitGroup{})
```

Comment:
Moving this outside of a goroutine stops cancellation via signal from working properly